src/components/variable.jsx

```diff
@@ -20,6 +25,29 @@ const Variable = React.createClass({
   getTodayInstant() {
     return new Date().toJSON().slice(0, 10)
   },
+  navigateProgramatically(e) {
```

Comment:
This seems complex. It seems that we are re-implementing routing. 

What problem that you encountered brought you to write this ? Did you get help from anyone who knows react well ?